src/conn_poll.c

```diff
@@ -402,21 +440,34 @@ int conn_poll_send(juice_agent_t *agent, const addr_record_t *dst, const char *d
 
 	if (conn_impl->send_ds >= 0 && conn_impl->send_ds != ds) {
 		JLOG_VERBOSE("Setting Differentiated Services field to 0x%X", ds);
-		if (udp_set_diffserv(conn_impl->sock, ds) == 0)
+		if (udp_set_diffserv(conn_impl->udp_sock, ds) == 0)
 			conn_impl->send_ds = ds;
 		else
 			conn_impl->send_ds = -1; // disable for next time
 	}
 
 	JLOG_VERBOSE("Sending datagram, size=%d", size);
 
-	int ret = udp_sendto(conn_impl->sock, data, size, dst);
+	int ret = 0;
+	if (dst->socktype == SOCK_STREAM) {
+		if (conn_impl->tcp_sock == INVALID_SOCKET) {
+			conn_impl->tcp_sock = tcp_create_socket(dst);
+			memcpy(&conn_impl->tcp_dst, dst, sizeof(conn_impl->tcp_dst));
+		}
+
+		ret = tcp_ice_write(conn_impl->tcp_sock, data, size);
```

Comment:
What if there are multiple TCP remote candidates? If called for different TCP `dst`, this will result in attempts to send to the first destination.

Reply:
Fixed. I ignore them at signaling time.

For the broadcast use case I only ever need one ice-tcp. In the future I could maybe add more?

What is the best resolution on complexity vs features on this one?